android/build.gradle

```diff
@@ -30,5 +39,6 @@ repositories {
 }
 
 dependencies {
-    compile 'com.facebook.react:react-native:+' // from node_modules
-}
\ No newline at end of file
+    //noinspection GradleDynamicVersion
+    implementation "com.facebook.react:react-native:${_reactNativeVersion}"
```

Comment:
I recommend replacing `implementation` with `compile`.

Reply:
`implementation` is the newest way to add dependencies. If you use `compile`, it throws a warning to replace it

Reply:
That may be true, but if you are using the default version on gradle that ships with React Native 0.56, the project will not run with `implementation`

Reply:
sorry but `implementation` must be used as by the end of 2018 old configs like `compile` would be removed completely

Reply:
I recommend you create a new react native project and then install and link react-native-config.  Running `react-native run-android` will fail to build if you use `implementation`.  It probably has something to do with the version of gradle react native uses out of the box. I'm just trying to keep this project compatible with React Native's default settings.

I'm not an Android developer, but I know what builds and what doesn't.  Maybe there is something else in my local environment that is preventing `implementation` from working.

In any event, I recommend testing the final PR with a brand new React Native project to ensure it works.

Reply:
Hi @taranda I know what you are trying to refer, I have a bit of experience with these gradle like issues. so the reason it will fail on new created project is that "Gradle config coming from react-native 0.56.0 is not Gradle 3+" thus we see that error. That error most likely will fix by react-native 0.57.0 because react-native must have to 💯 support Gradle 3+ as per Android new requirement, "Compile is obsolete and would be removed by the end of 2018" along all SDK new changes etc. Thus, we can either update now or wait for react-native to release 1 more new version with Gradle 3.

Reply:
That makes  sense.